src/utils/preprocessRequirement.ts

```diff
@@ -58,19 +64,22 @@ const preprocessRequirement = (requirement: Partial<Requirement>): Requirement =
     (processedRequirement.type === "ERC721" ||
       processedRequirement.type === "ERC1155" ||
       processedRequirement.type === "NOUNS") &&
-    !processedRequirement.data.attributes?.length &&
+    !processedRequirement?.data?.attributes?.length &&
     !processedRequirement.data?.ids?.length
   ) {
-    processedRequirement.data.attributes = undefined
-    if (!requirement.data.minAmount) processedRequirement.data.minAmount = 0
+    if (processedRequirement.data) {
+      processedRequirement.data.attributes = undefined
+      if (requirement.data && !requirement.data.minAmount)
+        processedRequirement.data.minAmount = 0
+    }
   }
 
   if (
     (processedRequirement.type === "ERC721" ||
       processedRequirement.type === "ERC1155") &&
     processedRequirement.data?.attributes?.length
   ) {
-    processedRequirement.data.attributes = requirement.data.attributes.map(
+    processedRequirement.data.attributes = requirement?.data?.attributes.map(
```

Comment:
This looks a bit strange, since we check if `processedRequirement.data.attributes.length` is greater than 0, so this array should be defined at this point.